### PR TITLE
Drag and drop support for color conditions

### LINKED ISSUE #480: Re-order color conditions
### What would you like to be added?

I want to be able to drag the different color condition lines set up in a Board view to re-order them. 

There could be a visual icon which allows you to drag the condition line up or down.

### Why is this needed?

Assigning colors to Board cards based on matching a certain condition is great. 

However, it seems that when a card matches two color conditions, it is the order in which the conditions themselves are created which defines which color gets assigned. Visually this is the top one in the condition list.

I am all for prioritizing the conditions with top priority at the top, but please let us rearrange the order by dragging the different condition lines. If I want to alter the color for a particular note matching multiple conditions, I have to start altering and deleting the conditions and that is a little counter-productive. Thanks!

### PR DESCRIPTION
**Pull request by Acylation**
fixes #480 

Major changes will be stressed in the following comments

https://github.com/marcusolsson/obsidian-projects/assets/73122375/a66192b5-749d-4195-9fd6-c62daebee9c5

### CODE REVIEW

File: src/components/FilterSettings/ColorFilterSettings.svelte [hunk 1]
```diff
@@ -34,6 +36,10 @@
     value: field.name,
   }));
 
+  type ColorRuleWithId = ColorRule & { readonly id: string };
+  let conditions: ColorRuleWithId[];
+  $: conditions = filter.conditions.map((cond) => ({ ...cond, id: uuidv4() }));
```
Comment (Acylation) [inline src/components/FilterSettings/ColorFilterSettings.svelte:41]: Add a new type `ColorRuleWithId` for both iteration and dnd, ids are assigned using `uuidv4()` **everytime** filter updates

File: src/components/FilterSettings/ColorFilterSettings.svelte [hunk 2]
```diff
@@ -205,44 +211,78 @@
...
+
+  const flipDurationMs = 200;
+
+  function handleDndConsider(e: CustomEvent<DndEvent<ColorRuleWithId>>) {
+    conditions = e.detail.items;
+  }
+
+  function handleDndFinalize(e: CustomEvent<DndEvent<ColorRuleWithId>>) {
+    filter = produce(filter, (draft) => {
+      draft.conditions = e.detail.items.map((cond) => ({
```
Comment (Acylation) [inline src/components/FilterSettings/ColorFilterSettings.svelte:223]: Remove `id` when finalize dnd action

File: src/components/FilterSettings/ColorFilterSettings.svelte [hunk 3]
```diff
@@ -205,44 +211,78 @@
...
-            value={condition.condition.value ?? ""}
-            on:blur={handleValueChange(i)}
-          />
-        {:else if isNumberFilterOperator(condition.condition.operator)}
-          <NumberInput
-            value={parseFloat(condition.condition.value ?? "")}
-            on:blur={handleValueChange(i)}
-          />
+  <div
+    style="display: flex; flex-direction: column; gap: 8px;"
```
Comment (Acylation) [inline src/components/FilterSettings/ColorFilterSettings.svelte:234]: Wrap-up conditions and exclude `Add Color` button from dnd area

File: src/components/FilterSettings/ColorFilterSettings.svelte [hunk 4]
```diff
@@ -205,44 +211,78 @@
...
+        transition: "all 150ms easy-in-out",
+      },
+    }}
+    on:consider={handleDndConsider}
+    on:finalize={handleDndFinalize}
+  >
+    {#each conditions as condition, i (condition.id)}
+      {@const field = getFieldByName(condition.condition.field)}
+      <HorizontalGroup>
+        <Icon name="grip-vertical" />
```
Comment (Acylation) [inline src/components/FilterSettings/ColorFilterSettings.svelte:252]: A handle for dnd

**Comment by marcusolsson** [comment]
Sorry for the long wait on this. I'll be reviewing this in the next coming days 🤞

**Comment by marcusolsson** [comment]
Looks and works beautiful! 👌 Thanks again for your high-quality PRs!

**Pull request merged**